frontend/src/metabase/collections/components/ActionMenu/ActionMenu.tsx

```diff
@@ -19,11 +19,20 @@ type Props = {
 };
 
 function getIsBookmarked(item: Item, bookmarks: Bookmarks) {
+  const normalizedItemModel = normalizeItemModel(item);
+
   return bookmarks.some(
-    bookmark => bookmark.type === item.model && bookmark.item_id === item.id,
+    bookmark =>
+      bookmark.type === normalizedItemModel && bookmark.item_id === item.id,
   );
 }
 
+// If item.model is `dataset`, that is, this is a Model in a product sense,
+// let’s call it "card" because `card` and `dataset` are treated the same in the back-end.
+function normalizeItemModel(item: Item) {
+  return item.model === "dataset" ? "card" : item.model;
```

Comment:
Should the backend use `card` instead of `dataset` in the first place? @kulyk WDYT?

Reply:
I thought they were going to add an additional `dataset` boolean field to Bookmark

Reply:
Anyway, the fix works

Reply:
> Should the backend use `card` instead of `dataset` in the first place? @kulyk WDYT?

Well, yeah, models are essentially cards with a `dataset: true` flag on the backend side. It'd be nice if a backend could figure out that "model: 'dataset'" should be treated as a card though